server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -355,6 +355,7 @@ public void applyClusterState(ClusterChangedEvent event) {
             }
             repositories = Collections.unmodifiableMap(builder);
         } catch (Exception ex) {
+            assert false : new AssertionError(ex);
```

Comment:
We do the same kind of assert up-stream in `org.elasticsearch.cluster.service.ClusterApplierService#runTask` but given #58680 I don't feel comfortable just yet to remove the try-catch-log` here.